src/compiler/crystal/command.cr

```diff
@@ -246,7 +251,11 @@ USAGE
     source = File.read(filename)
 
     begin
-      File.write(filename, Crystal::Formatter.format(source, filename: filename))
+      if @stdout
+        puts Crystal::Formatter.format(source, filename: filename)
```

Comment:
I suggest change this line similar to line 297 so code is formatted and then a single variable is used on the puts/write block:

```crystal
result = Crystal::Formatter.format(source, filename: filename)
if @stdout
  puts result
else
  File.write(filename, result)
end
```